code/stats/stats.cpp

```diff
@@ -13,253 +13,260 @@
+	{
+		if (type == StatsType::MISSION_STATS)
+		{
+			gr_printf_menu(sx, sy, "%s", XSTR("Player Deaths", 127));
+			sy += 2 * dy;
+		}
+
+		if (type == StatsType::MISSION_STATS)
+			gr_printf_menu(sx, sy, "%s", XSTR("Mission score", 1526));
+		else
+			gr_printf_menu(sx, sy, "%s", XSTR("Score", 1527));
+	}
+}
+
+void show_stats_numbers(StatsType type, int sx, int sy, int dy)
+{
+	switch (type)
+	{
+		case StatsType::MISSION_STATS:
+			show_stats_numbers(*Player_stats, true, sx, sy, dy);
+			break;
+
+		case StatsType::ALL_TIME_CAMPAIGN_STATS:
+			Assertion(!(Game_mode & GM_MULTIPLAYER), "ALL_TIME_CAMPAIGN_STATS are not stored for multiplayer!");
+			show_stats_numbers(*Player_stats, false, sx, sy, dy);
 			break;
 
-		case ALL_TIME_STATS:
-			gr_printf_menu(sx,sy,"%s", XSTR( "All Time Stats", 128));
-			sy += 2*dy;			
-			gr_printf_menu(sx,sy,"%s", XSTR( "Total kills", 115));
-			sy += 2*dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Primary weapon shots", 116));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Primary weapon hits", 117));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Primary friendly hits", 118));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Primary hit %", 119));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Primary friendly hit %", 120));
-			sy += 2*dy;
-
-			gr_printf_menu(sx,sy,"%s", XSTR( "Secondary weapon shots", 121));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Secondary weapon hits", 122));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Secondary friendly hits", 123));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Secondary hit %", 124));
-			sy += dy;
-			gr_printf_menu(sx,sy,"%s", XSTR( "Secondary friendly hit %", 125));
-			sy += 2*dy;			
-
-			gr_printf_menu(sx,sy,"%s", XSTR( "Assists", 126));
-			sy += 2*dy;
-
-			if(Game_mode & GM_MULTIPLAYER){
-				gr_printf_menu(sx,sy,"%s", XSTR( "Score", 1527));
-			}
+		case StatsType::ALL_TIME_EVER_STATS:
+			if (Game_mode & GM_MULTIPLAYER)
+				show_stats_numbers(*Player_stats, false, sx, sy, dy);		// in multiplayer, the all-time-ever stats are stored in the player
+			else
+				show_stats_numbers(All_time_ever_stats, false, sx, sy, dy);
 			break;
-		} // end switch
+
+		default:
+			UNREACHABLE("Unhandled StatsType %d in show_stats_numbers()", static_cast<int>(type));
+	}
 }
 
-void show_stats_numbers(int stage, int sx, int sy, int dy,int add_mission)
+int stats_get_kills(StatsType type, int ship_class)
+{
+	switch (type)
+	{
+		case StatsType::MISSION_STATS:
+			return stats_get_kills(*Player_stats, true, ship_class);
+
+		case StatsType::ALL_TIME_CAMPAIGN_STATS:
+			return stats_get_kills(*Player_stats, false, ship_class);
+
+		case StatsType::ALL_TIME_EVER_STATS:
+			if (Game_mode & GM_MULTIPLAYER)
+				return stats_get_kills(*Player_stats, false, ship_class);		// in multiplayer, the all-time-ever stats are stored in the player
+			else
+				return stats_get_kills(All_time_ever_stats, false, ship_class);
```

Comment:
Same thing here, just using the `return stats_get_kills(*Player_stats, false, ship_class);` line rather than needing an if-else.